src/hotspot/share/opto/loopTransform.cpp

```diff
@@ -1922,25 +1925,21 @@ void PhaseIdealLoop::update_main_loop_assertion_predicates(Node* ctrl, CountedLo
       break;
     }
     Node* bol = iff->in(1);
-    if (bol->is_Opaque4()) {
-      if (assertion_predicate_has_loop_opaque_node(iff)) {
-        // This is a Template Assertion Predicate for the initial or last access.
-        // Create an Initialized Assertion Predicates for it accordingly:
-        // - For the initial access a[init] (same as before)
-        // - For the last access a[init+new_stride-orig_stride] (with the new unroll stride)
-        prev_proj = create_initialized_assertion_predicate(iff, init, max_value, prev_proj);
-      } else {
```

Comment:
Why is it safe to remove this?

Reply:
Before the patch, we could have an `Opaque4` for an Template Assertion Predicate or Non-null check. Therefore, we additionally called `assertion_predicate_has_loop_opaque_node(iff)` to distinguish them. Now, it is enough to check for `bol->is_OpaqueTemplateAssertionPredicate()`. We can still have an `OpaqueNotNull` node which I now assert on L1942 instead.